Issue36 axis naming and stacking

julianhoersch opened the pull request:
This PR addresses #36 and introduces a couple of changes:

- The reference to any axes that do not change position and are essentially zero, but are not in `always_axes` is removed with prune_scan_info for cbf files
- The creation of detector axes has been changed, see [make_detector_axes](https://github.com/julianhoersch/instrument-geometry-info/blob/ec4b5141a12c635beea3847f60d8714547f34a3d/Tools/imgCIF_Creator/imgCIF_Creator/command_line_interfaces/parser.py#L208)
   - now multiple rotational and translational axes are allowed and automatically inserted into the axis chain
   - `detx`, `dety`, `source` and `gravity` are assumed to be always there. I assumed that the axis stacking must be something like: `dety`, `detx`, `trans_1`, `trans_2`, `rot_1`, `rot_2`, `gravity`, `source`
   - the axis type of the axis names found in the cbf file are identified as rotational or translational axes based on
   https://github.com/COMCIFS/instrument-geometry-info/blob/bc9059eca524b60fbe18080b85dc1daaace8ee8e/Tools/imgCIF_Creator/imgCIF_Creator/output_creator/imgcif_creator.py#L10-L14 
   - translational axes are added after `dety` with vector
    https://github.com/COMCIFS/instrument-geometry-info/blob/bc9059eca524b60fbe18080b85dc1daaace8ee8e/Tools/imgCIF_Creator/imgCIF_Creator/command_line_interfaces/parser.py#L261-L262
   - if a offset was found this will be added in the `z` component of offset otherwise it's just set to zero (currently also if you define new axes the offset will be 0)
   - rotational axes will be added subsequently
   - the rotational axis vector will be determined by the sense of rotation, first clockwise rotation is assumed (I guess there is no way of determining the rotation sense?) and then the user will be asked if this is alright
   - the axis `depends_on` field will be determined by the stacking of all those axes
   - _does that make sense?_
- the user can now edit the axis stacking and rotation senses of axes found in the cbf file and add new axes, see [complete goniometer axes](https://github.com/julianhoersch/instrument-geometry-info/blob/ec4b5141a12c635beea3847f60d8714547f34a3d/Tools/imgCIF_Creator/imgCIF_Creator/output_creator/imgcif_creator.py#L505) and [complete detector axes](https://github.com/julianhoersch/instrument-geometry-info/blob/ec4b5141a12c635beea3847f60d8714547f34a3d/Tools/imgCIF_Creator/imgCIF_Creator/output_creator/imgcif_creator.py#L560)
   - axes which were found in the cbf file will be shown with their stacking and rotation senses
   - the user can that decide to keep them as is, or edit the stacking, rotation senses or add new axes. The axes which are already found must be included in the new axes - this was done to keep the mapping with the file content. If it is desired to change axis names this should be done later maybe.
- if the name of the axis is not in the predefined, known axes and an `oscillation_axis` is defined in the cbf header then the original axis name as `anlge` will be replaced by the `oscillation_axis` name as `omega`, see [get_frame_info_cbf](https://github.com/julianhoersch/instrument-geometry-info/blob/ec4b5141a12c635beea3847f60d8714547f34a3d/Tools/imgCIF_Creator/imgCIF_Creator/information_extractors/cbf_smv.py#L775)

Please check if that makes sense and maybe just try it out by using my branch. It still asks for information of the detector axes, I think this will be redundant if everything is correct.

kroon-lab commented:
the rotational axis vector will be determined by the sense of rotation, first clockwise rotation is assumed (I guess there is no way of determining the rotation sense?) and then the user will be asked if this is alright
==> no, you can not know how the physical rotation axis is oriented. That is what the user has to tell you. So the user has to say if the axis rotates clockwise or anticlockwise (when looking from the sample into the goniometer)

julianhoersch commented:
Ok. So then the part about questioning for the rotation sense is fine. 

I've just realized (see also #46) that axes that are defined by the user will not show up in the scan list, so I guess that would be a problem for cbflib. Should they better appear and we add some zeros for the scan start and range or ask the user for more information? or should we leave axes that are not scanned entirely out of the scan description?

julianhoersch commented:
I have removed the option to add new axes to the axes which were already found. Now chi or kappa are only requested if there is an axis which contains `chi` or `kappa` in its name and kappa alows only 180 or 0 as second angle. The detector translation axis is also only displayed if there is more than one, since you can't cange anything if it's one. Axes which do not move and are not at their default positions are now also removed from hdf5 files.
